src/dali_executor/dali_pipeline.h

```diff
@@ -110,10 +110,14 @@ class DaliPipeline {
       dali_data_type_t data_type, span<const int64_t> inputs_shapes,
       int sample_ndims);
 
+  void SetInput(
+      const void* ptr, const char* name, device_type_t source_device,
+      dali_data_type_t data_type, TensorListShape<> input_shape);
+
   void PutOutput(
       void* destination, int output_idx, device_type_t destination_device);
 
-  const int batch_size;
+  const int max_batch_size;
```

Comment:
This is movable (but it doesn't have a move assignment), and probably one reason of this is the const member. Some people discourage using them.

Also, the default constructor (or rather one arg constructor with default argument max_batch_size=0 creates a rather useless pipeline).

Reply:
I changed the default constructor to create empty handle that does not create stream, nor initialize DALI

Reply:
Also added move assign